index.js

```diff
@@ -179,9 +179,18 @@ const plugin = fp(async function (app, opts) {
     if (gateway.pollingInterval !== undefined) {
       if (typeof gateway.pollingInterval === 'number') {
         gatewayInterval = setInterval(async () => {
-          const schema = await gateway.refresh()
-          if (schema !== null) {
-            fastifyGraphQl.replaceSchema(schema)
+          try {
+            const context = assignApplicationLifecycleHooksToContext(fastifyGraphQl[kHooks], {})
+            const schema = await gateway.refresh()
+            if (schema !== null) {
+              // Trigger onGatewayReplaceSchema hook
+              if (context.onGatewayReplaceSchema !== null) {
+                await onGatewayReplaceSchemaHandler(context, { instance: app, schema })
+              }
+              fastifyGraphQl.replaceSchema(schema)
+            }
+          } catch (error) {
```

Comment:
Drawing attention to this. I added the catch and subsequent log because when testing hooks error handling, I was getting an unhandled Promise rejection because a promise is passed to setInterval and this situation was previously untested (we would get the same problem if `gateway.refresh` errored).